platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/AbstractIdeSyncSmokeTest.groovy

```diff
@@ -0,0 +1,182 @@
+        )
+    }
+
+    protected void ideaSync(String ideaHome) {
+        def invocationSettings =
+            syncInvocationSettingsBuilder()
+                .setStudioInstallDir(new File(ideaHome))
+                .build()
+
+        sync(
+            "IDEA sync",
+            invocationSettings,
+            Collections.emptyList()
+        )
+    }
+
+    private InvocationSettings.InvocationSettingsBuilder syncInvocationSettingsBuilder() {
+        // TODO it would be nice not to have it in test dir, to avoid it's indexing during a sync
+        def gradleUserHome = file("gradle-user-home")
+        def ideSandbox = file("ide-sandbox")
+        def profilerOutput = file('profiler-output')
+
+        // TODO -Dstudio.tests.headless=true check headless mode possibility
+        return new InvocationSettings.InvocationSettingsBuilder()
+            .setProjectDir(testDirectory)
+            .setProfiler(Profiler.NONE)
+            .setStudioSandboxDir(ideSandbox)
+            .setGradleUserHome(gradleUserHome)
+            .setVersions(ImmutableList.of(distribution.version.version))
+            .setScenarioFile(null)
+            .setBenchmark(true)
+            .setOutputDir(profilerOutput)
+            .setWarmupCount(1)
+            .setIterations(1)
+    }
+
+    private void sync(
+        String scenarioName,
+        InvocationSettings invocationSettings,
+        List<BuildMutator> buildMutators
+    ) {
+        // TODO consider passing jvmArgs where it's sane
+        def syncScenario = new StudioGradleScenarioDefinition(
+            new GradleScenarioDefinition(
+                scenarioName,
+                scenarioName,
+                GradleBuildInvoker.AndroidStudio,
+                new GradleBuildConfiguration(
+                    distribution.getVersion(),
+                    distribution.gradleHomeDir,
+                    Jvm.current().getJavaHome(),
+                    Collections.emptyList(), // daemon args
+                    false,
+                    Collections.emptyList() // client args
+                ),
+                new AndroidStudioSyncAction(),
+                BuildAction.NO_OP,
+                Collections.emptyList(),
+                new HashMap<String, String>(),
+                buildMutators,
+                invocationSettings.warmUpCount,
+                invocationSettings.buildCount,
+                invocationSettings.outputDir,
+                Collections.emptyList(),
+                Collections.emptyList()
+            ),
+            Collections.emptyList(),
+            Collections.emptyList()
+        )
+
+        def scenarioInvoker = new StudioGradleScenarioInvoker(
+            new GradleScenarioInvoker(
+                new DaemonControl(invocationSettings.getGradleUserHome()),
+                createPidInstrumentation()
+            )
+        )
+
+        Logging.setupLogging(invocationSettings.outputDir)
```

Comment:
That call is bypassed when scenario is invoked directly (as we doing), but it's invoked in regular Profiler flow. This is allowing to write profiler logs to `profile.log`. It can be useful I think